staging/src/k8s.io/client-go/tools/cache/listwatch.go

```diff
@@ -63,8 +63,18 @@ type Getter interface {
 
 // NewListWatchFromClient creates a new ListWatch from the specified client, resource, namespace and field selector.
 func NewListWatchFromClient(c Getter, resource string, namespace string, fieldSelector fields.Selector) *ListWatch {
-	listFunc := func(options metav1.ListOptions) (runtime.Object, error) {
+	optionsModifier := func(options *metav1.ListOptions) {
 		options.FieldSelector = fieldSelector.String()
+	}
+	return NewListWatchFromClientWithModifier(c, resource, namespace, optionsModifier)
+}
+
+// NewListWatchFromClientWithSelectors creates a new ListWatch from the specified client, resource, namespace, and option modifier.
+// Option modifier is a function takes a ListOptions and modifies the consumed ListOptions. Provide customized modifier function
+// to apply modification to ListOptions with a field selector, a label selector, or any other desired options.
+func NewListWatchFromClientWithModifier(c Getter, resource string, namespace string, optionsModifier func(options *metav1.ListOptions)) *ListWatch {
```

Comment:
To be inline with the naming of filtered informers (`NewFilteredSharedInformerFactory`), let's call it `NewFilteredListWatchFromClient`.

Reply:
otherwise lgtm